### PR TITLE
Sliding Sync: Split and move tests

### PR DESCRIPTION
**Pull request by MadLittleMods**
Split and move Sliding Sync tests so we have some more sane test file sizes

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] [Code style](https://element-hq.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: tests/rest/client/sliding_sync/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#
+# See the GNU Affero General Public License for more details:
+# <https://www.gnu.org/licenses/agpl-3.0.html>.
+#
+# Originally licensed under the Apache License, Version 2.0:
+# <http://www.apache.org/licenses/LICENSE-2.0>.
+#
+# [This file includes modifications made by New Vector Limited]
+#
+#
```
Comment (MadLittleMods) [inline tests/rest/client/sliding_sync/__init__.py:20]: What's the correct copyright header for all of these?

Comment (erikjohnston) [inline tests/rest/client/sliding_sync/__init__.py:20]:
```suggestion
#
# This file is licensed under the Affero General Public License (AGPL) version 3.
#
# Copyright (C) 2024 New Vector, Ltd
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU Affero General Public License as
# published by the Free Software Foundation, either version 3 of the
# License, or (at your option) any later version.
#
# See the GNU Affero General Public License for more details:
# <https://www.gnu.org/licenses/agpl-3.0.html>.
#
```

is the correct header I think

**Comment by MadLittleMods** [comment]
> (I'm assuming that none of the tests have actually changed)

Just a few minor things I saw while splitting things. For example, stop requesting `required_state` for the `timeline` tests.

**Comment by MadLittleMods** [comment]
Thanks for the review @erikjohnston 🐛

**Pull request merged**